### PR TITLE
Feature/search page

### PR DESCRIPTION
**Pull request by MittBrukernavn**
Added a search page. Does not support tags. Currently does filtering frontend, since the search API only exists for metadata types.

### CODE REVIEW

**Comment by eliasvagan** [comment]
Are we waiting for tag search support before merging this?

**Comment by MittBrukernavn** [comment]
No, that was mostly to clarify that it doesn't filter by it as is. Night be worth considering to remove it entirely just to avoid confusion. The filtering should be done backend when support for that exists, where it should be easier. To do this frontend you would need to make an additional GET request for all the results to actually see tags, which would be slow and resource-demanding at best

**Comment by AndreasJJ** [comment]
Should we make a decision on removing it entirely or not?

File: frontend/src/pages/Search/SearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    align-items: center;
+    & > ul > li {
+      margin-top: 0px;
+    }
+  }
+`;
+
+export const SearchBar = ({ setQuery }) => {
+  const nameField = useRef('');
+  const descriptionField = useRef('');
```
Comment (AndreasJJ) [inline frontend/src/pages/Search/SearchBar.jsx]: Why are we using refs and not state?

It adds unnecessary overhead as far as I can see as we never actually use the ref updates. This is better solved with onChange and useState if I am not missing something

Comment (MittBrukernavn) [reply frontend/src/pages/Search/SearchBar.jsx]: The main reason I chose useRef is that useState/onChange forces re-render on every input, which should really only be necessary when the search button is clicked anyways. I can change if you think the cons of useRef outweigh the pros.

**Pull request merged**